met-api/tests/unit/services/test_keycloak.py

```diff
@@ -16,38 +16,135 @@
 
 Test-Suite to ensure that the Keycloak Service is working as expected.
 """
-# from met_api.services.keycloak import KeycloakService
-# from tests.utilities.factory_scenarios import KeycloakScenario
-#
-# KEYCLOAK_SERVICE = KeycloakService()
-
-# TODO: Replace this test with one that gets user composite roles
-# def test_keycloak_add_user(session):
-#     """Add user to Keycloak. Assert return a user with the same username as the username in request."""
-#     request = KeycloakScenario.create_user_request()
-#     user = KEYCLOAK_SERVICE.add_user(request)
-#     assert user.get('username') == request.get('username')
-
-# TODO: Replace this test with one that gets user composite roles
-# def test_keycloak_get_user_by_username(session):
-#     """Get user by username. Assert get a user with the same username as the username in request."""
-#     request = KeycloakScenario.create_user_request()
-#     KEYCLOAK_SERVICE.add_user(request)
-#     user = KEYCLOAK_SERVICE.get_user_by_username(request.get('username'))
-#     assert user.get('username') == request.get('username')
-
-# TODO: Replace this test with one that gets user composite roles
-# def test_keycloak_get_user_groups(session):
-#     """Get user by username. Assert get a user with the same username as the username in request."""
-#     request = KeycloakScenario.create_user_request()
-#     group_name = 'admins'
-#     KEYCLOAK_SERVICE.add_user(request)
-#     user = KEYCLOAK_SERVICE.get_user_by_username(request.get('username'))
-#     user_id = user.get('id')
-#     user_group = KEYCLOAK_SERVICE.get_users_groups([user_id])
-
-#     assert group_name not in user_group.get(user_id)
-#     # add the group
-#     KEYCLOAK_SERVICE.add_user_to_group(user_id, '%s' % group_name)
-#     user_group = KEYCLOAK_SERVICE.get_users_groups([user_id])
-#     assert group_name in user_group.get(user_id)
+import pytest
+import requests
+from met_api.services.keycloak import KeycloakService
+from unittest.mock import patch
+
+
+KEYCLOAK_GET_REQUEST = 'met_api.services.keycloak.requests.get'
+KEYCLOAK_POST_REQUEST = 'met_api.services.keycloak.requests.post'
+KEYCLOAK_DEL_REQUEST = 'met_api.services.keycloak.requests.delete'
+KEYCLOAK_SERVICE = KeycloakService()
+
+
+@pytest.fixture()
```

Comment:
Is this patch intended to run before each function? If so, maybe explicitly specify `fixture(scope='function')`. If not, specify another scope such as `module`

Reply:
@NatSquared  Thanks for notifying this. When I checked the default scope of the fixture will be **function** always. We need that because we have a test named def **test_get_admin_token**(mock_post) that checks whether _get_admin_token is called once or not. Also, all other test functions require this patch. That is why I left it to default scope which is function.